### PR TITLE
Visual Design and Hierarchy with ElkJS

### PR DESCRIPTION
**Pull request by clark-pang**
Implement DAG auto distributed layout from initialized React Flow nodes and edges by utilizing ElkJS
Create dynamic handles that react to positioning and direction
Create baseline 'breathable' visual flow
NOTE: Loading new endpoint after initial WILL break the app at this point. You must refresh before loading new endpoint. This will be fixed later. The issue is race conditions, multiple useEffects, and resolving artifacts of previous nodes and edges before regraphing.

### CODE REVIEW

File: client/src/components/Field.jsx [hunk 1]
```diff
@@ -17,38 +18,68 @@ const fieldData = {
...
+    const targetPosition = targetNode.position;
+    const currPosition = currNode.position;
+    if (currPosition.x > targetPosition.x && handlePosition !== 'left') {
+      setHandlePosition('left');
+      updateNodeInternals(typeName);
+    } else if (currPosition.x < targetPosition.x && handlePosition !== 'right') {
+      setHandlePosition('right');
+      updateNodeInternals(typeName);
+    }
+  }
```
Comment (gacetta) [inline client/src/components/Field.jsx:72]: This is awesome.  Totally how I was thinking through doing this in the shower today!

File: client/src/components/Field.jsx [hunk 2]
```diff
@@ -17,38 +18,68 @@ const fieldData = {
...
+  // and assign them only once in useEffect, however ... that created
+  // all sorts of unintended behavior
+  // You'd think it'd be easier that way ... it seems 'references' got lost in state
+  // So here, we're 'brute forcing' instead.
+  if (relationship) {
+    const targetNode = nodes.find(node => node.id === relationship);
+    const currNode = nodes.find(node => node.id === typeName);
+    const targetPosition = targetNode.position;
+    const currPosition = currNode.position;
+    if (currPosition.x > targetPosition.x && handlePosition !== 'left') {
```
Comment (gacetta) [inline client/src/components/Field.jsx:65]: one note / question without seeing it: if edges properly avoid nodes, do we need to factor some sort of "gapSize" into this check to let the edge through before the switch?  

Maybe silly thought - just popped in my head

**Review by gacetta — APPROVED** [review]
I'll need to look into elkJS more to understand.  

smart handle switcher is great!  Looking forward to seeing everything.

File: client/src/components/Field.jsx (hunk 2 above)
Comment (clark-pang) [reply client/src/components/Field.jsx:65]: Atm, edges are not avoiding nodes -- that is a complex feature I worked on but realized it was going to take too much, so I'm waiting to come back to it later :)

**Pull request merged**